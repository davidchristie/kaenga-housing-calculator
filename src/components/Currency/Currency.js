/* eslint-disable react/style-prop-object */

import React, { Component, PropTypes } from 'react'
import { FormattedNumber } from 'react-intl'

class Currency extends Component {
  render () {
    const { type, value } = this.props
    return (
      <FormattedNumber currency={type} style='currency' value={value} />
    )
  }
}

Currency.defaultProps = {
  type: 'nzd',
  value: 0
}

Currency.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired
}

export default Currency
