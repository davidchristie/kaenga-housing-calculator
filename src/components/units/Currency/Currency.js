import React, { Component, PropTypes } from 'react'
import { FormattedNumber } from 'react-intl'

class Currency extends Component {
  render () {
    const { value } = this.props
    if (value > 1000) {
      return (
        <span>
          $
          <FormattedNumber
            maximumSignificantDigits={2}
            value={value / 1000}
          />
          K
        </span>
      )
    } else {
      return (
        <span>
          $
          <FormattedNumber
            maximumSignificantDigits={2}
            value={value}
          />
        </span>
      )
    }
  }
}

Currency.defaultProps = {
  value: 0
}

Currency.propTypes = {
  value: PropTypes.number.isRequired
}

export default Currency
