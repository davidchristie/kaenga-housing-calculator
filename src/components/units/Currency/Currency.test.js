import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import Currency from './Currency'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Currency />)).toMatchSnapshot()
})
