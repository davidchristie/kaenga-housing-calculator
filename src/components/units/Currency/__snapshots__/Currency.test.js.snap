exports[`test matches snapshot 1`] = `
<span>
  $
  <FormattedNumber
    maximumSignificantDigits={2}
    value={0} />
</span>
`;
